### PR TITLE
Refactor: `AuthProvider` to `ClaimsProvider`

### LINKED ISSUE #2233: Model refactor: `AuthProvider`
## Acceptance criteria

- [x] Model renamed to `ClaimsProvider`
- [x] Each field in the model definition has a description
- [x] Field `EligibilityVerifier.auth_provider` renamed to `EligibilityVerifier.claims_provider`
- [x] All variable instances and functions renamed to `claims` or `claims_provider` (e.g. `uses_claims_verification` instead of `uses_auth_verification`)
- [x] All comments updated to remove mention of "auth" or "auth provider", using "claims" or "claims provider" instead
- [x] Existing Django config data is migrated
- [x] Docs are updated

## Impact on the code

- [ ] [Code search for `AuthProvider` in the repo](https://github.com/search?q=repo%3Acal-itp%2Fbenefits+AuthProvider&type=code)
- [ ] [Code search for `auth_provider` in the repo](https://github.com/search?q=repo%3Acal-itp%2Fbenefits+auth_provider&type=code)
- [ ] [Code search for `auth_required` in the repo](https://github.com/search?q=repo%3Acal-itp%2Fbenefits+auth_required&type=code)
- [ ] [Code search for `auth_verification` in the repo](https://github.com/search?q=repo%3Acal-itp%2Fbenefits+auth_verification&type=code)
- [ ] [Code search for `client_name` in the repo](https://github.com/search?q=repo%3Acal-itp%2Fbenefits+client_name&type=code)

## Impact on existing data

- [x] Standard migrations for model/field renames should maintain existing data

## Impact on analytics

- [ ] [The `client_name` field is logged in analytics by setting the `auth_provider` event property](https://github.com/cal-itp/benefits/blob/main/benefits/oauth/analytics.py#L15). Since this field will be renamed in the code, we should rename the property in analytics.
- [ ] #2247 is a follow-up

## Impact on Admin permissions

- [x] [View permissions](https://github.com/cal-itp/benefits/blob/main/benefits/core/migrations/0014_staff_group_view_permissions.py) need to be updated for the new model name
- [x] [Edit permissions](https://github.com/cal-itp/benefits/blob/main/benefits/core/migrations/0015_staff_group_edit_permissions.py) need to be updated for the new model name
- [x] [Read only fields](https://github.com/cal-itp/benefits/blob/main/benefits/core/admin.py#L83) need to be updated with the renamed field

### PR DESCRIPTION
**Pull request by angela-tran**
Closes #2233 - see Acceptance Criteria on that ticket.

## Testing locally

1. Checkout the `main` branch
1. Make sure your `local_fixtures.json` (or whatever fixture you are using) has an `AuthProvider` and an `EligiblityVerifier` that has a foreign key reference to that auth provider
1. Run `bin/reset_db.sh` (now your local DB mirrors the structure of the `dev` environment)
1. Checkout this PR's branch
1. Run `bin/init.sh` (running this PR's migration on top of existing DB structure)

- [ ] For all `EligibilityVerifier`s that with a reference to an `AuthProvider`, confirm those references were maintained, only now the field is called "Claims provider" in the admin interface, `claims_provider` in the database
- [ ] Confirm the `Can X auth provider` permissions are removed from the database
- [ ] Confirm the `Can X claims provider` permissions are added to the database
- [ ] Confirm the Cal-ITP group has `Can [view|change] claims provider` permissions
- [ ] Flows using claims from IdG should still work

### CODE REVIEW

**Comment by angela-tran** [comment]
Ready for review

**Comment by angela-tran** [comment]
Not sure if this one (`authentication providers`) needs to be updated and/or what the right wording would be.

https://github.com/cal-itp/benefits/blob/58abd80eed9e22cc71cadcd4d45059919370ec72/docs/README.md?plain=1#L41

**Comment by thekaveman** [comment]
> Not sure if this one (`authentication providers`) needs to be updated and/or what the right wording would be.
> 
> https://github.com/cal-itp/benefits/blob/58abd80eed9e22cc71cadcd4d45059919370ec72/docs/README.md?plain=1#L41

I think `claims providers` here is fine?

**Review by thekaveman — CHANGES_REQUESTED** [review]

**Review by lalver1 — COMMENTED** [review]
The code changes look good! These were verified:

- [x] For all `EligibilityVerifier`s that with a reference to an `AuthProvider`, confirm those references were maintained, only now the field is called "Claims provider" in the admin interface, `claims_provider` in the database
- [x] Confirm the `Can X auth provider` permissions are removed from the database
- [x] Confirm the `Can X claims provider` permissions are added to the database
- [x] Confirm the Cal-ITP group has `Can [view|change] claims provider` permissions
- [x] Flows using claims from IdG should still work

Just saw a minor typo in `oauth.md`.

File: docs/configuration/oauth.md [hunk 1]
```diff
@@ -22,9 +22,9 @@ for more details about what features are available. Specifically, from Authlib w
 
 ## Django configuration
 
-OAuth settings are configured as instances of the [`AuthProvider` model](../development/models-migrations.md).
+OAuth settings are configured as instances of the [`ClaimsProvider` model](../development/models-migrations.md).
 
-The [data migration file](./data.md) contains sample values for an `AuthProvider` configuration. You can set values for a real Open ID Connect provider in environment variables so that they are used instead of the sample values.
+The [data migration file](./data.md) contains sample values for an `ClaimsProvider` configuration. You can set values for a real Open ID Connect provider in environment variables so that they are used instead of the sample values.
```
[the remark this answers is not in the document]
Comment (angela-tran) [reply docs/configuration/oauth.md]: Thanks! Made that change in [ee3df8ec](https://github.com/cal-itp/benefits/pull/2256/commits/ee3df8ec8bdda588730e10a2e62b28a88202cd5b#diff-6339472682ab7bbe740e47e1193103b337eb1babcc56266cd838a9ca5c051b82R27)

**Review by thekaveman — APPROVED** [review]
This looks great :+1: !!!

**Pull request merged**